upgrade_logic/base_logic.py

```diff
@@ -79,3 +81,13 @@ def call_function_by_name(function_name: str) -> bool:
         return f_lookup[function_name]()
     print("function " + function_name + " not found", flush=True)
     return False
+
+
+# function only sets the versions, not the update login in between
+def update_versions_to_newest() -> None:
+    services = app_version.get_all()
+    for service in services:
+        if service.installed_version != service.remote_version:
+            app_version.update(
+                service.service, service.remote_version, with_commit=True
```

Comment:
without commit in the update and in the end (after the for loop) and if something was changed a general.commit()

Reply:
also why is this done like this (select all, loop over array and change every single one)? 
We can solve this without collecting the data to python with an SQL query.
```
UPDATE app_version
SET installed_version = remote_version
WHERE installed_version != remote_version
```
should do the trick